### ISSUE TITLE
Multiple markers not handled correctly

### ISSUE THREAD

**Issue by danjo133** [osiegmar/logback-gelf#35]
Hi,

I'm having some problems with the gelf output, if I understand correctly, this line should loop over related markers and add them as properties like Mdc does. 

https://github.com/osiegmar/logback-gelf/blob/97a14b719c3ceb8882396b610ae96326b7a3a32e/src/main/java/de/siegmar/logbackgelf/GelfEncoder.java#L337

With multiple markers I get a _marker="LS_APPEND_OBJECT" in my json output.

For example logstash logback adds the various markers as separate properties

Best regards,
Daniel

**Comment by osiegmar** [comment]
Indeed, multiple markers are not yet supported. Could you give an example of logstash handles multiple markers / properties?

**Comment by danjo133** [comment]
When you add multiple markers with and, they get added as a reference, from
https://github.com/logstash/logstash-logback-encoder/blob/master/src/main/java/net/logstash/logback/marker/LogstashMarker.java


```
    /**
     * Adds the given marker as a reference, and returns this marker.
     * <p>
     * This can be used to chain markers together fluently on a log line. For example:
     * 
     * <pre>
     * {@code
     * import static net.logstash.logback.marker.Markers.*
     *     
     * logger.info(append("name1", "value1).and(append("name2", "value2")), "log message");
     * }
     * </pre>
     * 
     * @param <T> subtype of LogstashMarker
     * @param reference The marker to add
     * @return A marker with this marker and the given marker
     */
    @SuppressWarnings("unchecked")
    public <T extends LogstashMarker> T and(Marker reference) {
        add(reference);
        return (T) this;
    }
```
Then they can be added as a string as they are in that file

   ```
 public String toString() {
        String self = toStringSelf();
        if (!hasReferences()) {
            return self;
        }

        StringBuilder sb = new StringBuilder(self);
        boolean appendSeparator = !self.isEmpty();
        for (Marker marker : this) {
            if (appendSeparator) {
                sb.append(", ");
            }
            String referenceToString = marker.toString();
            sb.append(referenceToString);
            appendSeparator = !referenceToString.isEmpty();
        }

        return sb.toString();
    }
```

or as json properties as in:

https://github.com/logstash/logstash-logback-encoder/blob/master/src/main/java/net/logstash/logback/marker/MapEntriesAppendingMarker.java

```
    @Override
    public void writeTo(JsonGenerator generator) throws IOException {
        if (map != null) {
            for (Map.Entry<?, ?> entry : map.entrySet()) {
                generator.writeFieldName(String.valueOf(entry.getKey()));
                generator.writeObject(entry.getValue());
            }
        }
    }
```
Which is then added the same way as mdc properties.

**Comment by osiegmar** [comment]
I've added support for multiple markers in the [bugfix/multiple-markers](https://github.com/osiegmar/logback-gelf/tree/bugfix/multiple-markers) branch. Could you please check and confirm, that this also fixes your issue? I'm not sure where your LS_APPEND_OBJECT comes from.

**Comment by osiegmar** [comment]
Fix released in 2.2.0.
